cosmoz-default-tree.html

```diff
@@ -23,134 +34,192 @@
-			return;
+	/**
+	 * Returns the node of a given path.
+	 * @return {object}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj (The object the search should be based on.) Default: this._treeData
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	 * @param {string} childProperty (The string which describes the child property of a node.) Default: "children"
+	*/
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator, nodeObj, separatorSign, childProperty) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+		// Defaults
+		nodeObj = nodeObj ? nodeObj : this._treeData;
+		separatorSign = separatorSign ? separatorSign : '.';
+		childProperty = childProperty ? childProperty : 'children';
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+		var	pathNodes = this.getPathNodes(pathLocator, nodeObj, separatorSign, childProperty),
+			lastNodeOnPath = pathNodes.pop();
+		return lastNodeOnPath !== undefined ? lastNodeOnPath : null;
 	};
 
-	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
-		if (node === undefined) {
-			return;
+	/**
+	 * Returns the nodes on a given path.
+	 * @return {array} or null if there was an empty response.
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj (The object the search should be based on.) Default: this._treeData
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	 * @param {string} childProperty (The string which describes the child property of a node.) Default: "children"
+	*/
+	Cosmoz.DefaultTree.prototype.getPathNodes = function (pathLocator, nodeObj, separatorSign, childProperty) {
+		if (!pathLocator) {
+			return this._roots;
 		}
-		return _objectValues(node.children);
-	};
 
-	Cosmoz.DefaultTree.prototype.getProperty = function (node, propertyName) {
-		if (node === undefined || propertyName === undefined) {
-			return;
-		}
+		// Defaults
+		separatorSign = separatorSign ? separatorSign : '.';
+		childProperty = childProperty ? childProperty : 'children';
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodeObj ? nodeObj : this._treeData,
+			pathIsInvalid,
+			nodes;
+		
+		// Get the nodes on the path.
+		nodes = path.map(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = node[childProperty];
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			}
+			return node;
+		}, this);
 
-		return node[propertyName];
-	};
+		// Check if all nodes on the path are valid.
+		pathIsInvalid = nodes.some(function (n) {
```

Comment:
I'm not sure...

Imagine:
```
{
  100: {children: {9: {children: {7: {"code": "secret"}}}}},
  9: {children: {7: {"code": "hello"}}},
}
```
A user calls `getPathNodes()`with an invalid path `99.9.7` instead of `100.9.7`

the function returns:
```
 var result = getPathNodes('99.9.7') 
// returns [9, 7] => result[1].code == 'hello' instead of 'secret'
```
How can a user validate the return? 

a) If we filter out undefined values: By `return.length === path.length`

b) If we leave them: By  `return.every(n => n)`

c) If we return `null`: Just by checking `if (return)`

I think I would still go with c) because you'd need to check each time after you call the function to be consistent. Meaning, I think we should return `null` if the path is not 100% valid.

What do you think?